mocks/mocks.py

```diff
@@ -95,39 +256,44 @@ def scale_light_by_distance(particle_positions,
             particle_positions,
             particle_values,
             comoving_correction_length,
-            viewing_axis=viewing_axis)
+            viewing_axis=config['viewing_axis'])
     # Converting co-moving distances to luminosity distances.
     particle_positions = convert_to_lumonsity_distance(particle_positions,
-                                                       viewing_axis,
+                                                       config['viewing_axis'],
                                                        cosmo)
     # Applying corrections to the mass weights of particles.
     particle_values =\
         particle_values / (particle_positions[:, AxisNumber[
-                           viewing_axis]] / physical_distance) ** 2
+            config['viewing_axis']]] / config['galaxy_distance']) ** 2
 
     return particle_positions.value, particle_values
 
 
-def prepare_mocks(config_location='config_example.yaml',
+def prepare_mocks(config,
                   **kwargs):
     """
     Creates a dictionary containing configuration data and a numpy.array of
     the simulation data.
 
     Parameters
     ----------
-    config_location : str, optional
-        The name (location) of the yaml file that contains initial
-        information, default `config_example.yaml`.
+    config : dict
+        A dictionary of configuration items.
+    **kwargs
+        User can add the galaxy coordinates and observation time by kwargs.
```

Comment:
Should not be able to. As in comment above, the only place to override configuration parameters should be `parse_config()` in order to ensure a consistent set of config parameters are used by all the functions.